From 2cf3af69eba66f7e36b9d1ec930e2e03b7edd2bf Mon Sep 17 00:00:00 2001
From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Wed, 10 Dec 2014 10:29:28 -0500
Subject: [wireless] rt2800: correct BBP1_TX_POWER_CTRL mask

Message-id: <1418207389-7899-15-git-send-email-sgruszka@redhat.com>
Patchwork-id: 101584
O-Subject: [RHEL7.1 PATCH 14/35] rt2800: correct BBP1_TX_POWER_CTRL mask
Bugzilla: 1155538
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

BZ#1155538

commit 01f7feeaf4528bec83798316b3c811701bac5d3e upstream.

Two bits control TX power on BBP_R1 register. Correct the mask,
otherwise we clear additional bit on BBP_R1 register, what can have
unknown, possible negative effect.

Signed-off-by: Stanislaw Gruszka <sgruszka@redhat.com>
Signed-off-by: John W. Linville <linville@tuxdriver.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Brew:
https://brewweb.devel.redhat.com/taskinfo?taskID=8330268

Testing:
Tested on RT3062 PCI and RT3070 USB adapters.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/wireless/rt2x00/rt2800.h b/drivers/net/wireless/rt2x00/rt2800.h
index a394a9a..7cf6081 100644
--- a/drivers/net/wireless/rt2x00/rt2800.h
+++ b/drivers/net/wireless/rt2x00/rt2800.h
@@ -2039,7 +2039,7 @@ struct mac_iveiv_entry {
  * 2 - drop tx power by 12dBm,
  * 3 - increase tx power by 6dBm
  */
-#define BBP1_TX_POWER_CTRL  FIELD8(0x07)
+#define BBP1_TX_POWER_CTRL  FIELD8(0x03)
 #define BBP1_TX_ANTENNA   FIELD8(0x18)
 
 /*
-- 
1.7.1